### PR TITLE
I2C give each slave access to their own buffer

### PR DESCRIPTION
**Pull request by RuthgerD**
depends on #49

### CODE REVIEW

**Comment by RuthgerD** [comment]
Once SPI is merged add `copy_proto_slaves(&BoardData::spi_buses)` for merge_board support

File: modules/ardrivo/src/Entrypoint.cxx [hunk 1]
```diff
@@ -80,7 +80,12 @@ void merge_boards(BoardData old_board, BoardData& new_board) noexcept {
...
             std::tie(to.rx, to.tx) = std::forward_as_tuple(from.rx, from.tx);
     };
-
-    copy_proto_bufs(&BoardData::i2c_buses);
     copy_proto_bufs(&BoardData::uart_buses);
+
+    auto copy_proto_slaves = [&]<class T>(std::vector<T>(BoardData::* mvec)) {
+        for(auto [from, to] : ranges::views::zip(old_board.*mvec, new_board.*mvec))
+            to.slaves = std::move(from.slaves);
+    };
```
Comment (AeroStun) [inline modules/ardrivo/src/Entrypoint.cxx:88]: Wouldn't [`ranges::move`](https://ericniebler.github.io/range-v3/group__group-algorithms.html#gaa55485146994b298d1392f1aa421fdfe) be better?

Comment (AeroStun) [reply modules/ardrivo/src/Entrypoint.cxx:88]: NVM it's not due to possibly mismatching sizes

File: modules/ardrivo/src/Wire.cxx [hunk 2]
```diff
@@ -23,84 +22,80 @@ void TwoWire::begin(int sda, int scl) {
...
-                       return std::move(lock);
+    std::visit(Visitor{
+                   [quantity, &lock](BlockingBus<std::tuple<>>& blocking_bus) mutable {
+                       blocking_bus.request_bytes = quantity;
+                       blocking_bus.sync.wait(lock);
                    },
-                   [=, lock = std::move(lock)](const std::function<void(std::size_t)>& generator) mutable { generator(quantity); return std::move(lock); },
-               }, device);
-    return board_data->i2c_buses[bus_id].rx_used_size;
+                   [=, &lock](const std::function<void(std::size_t)>& generator) mutable { generator(quantity); },
```
Comment (AeroStun) [inline modules/ardrivo/src/Wire.cxx:63]: Why do you capture the lock?

**Pull request merged**